src/Miunie.Storage/JsonPersistentStorage.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Miunie.Core.Logging;
+using Miunie.Core.Storage;
+using Newtonsoft.Json;
+
+namespace Miunie.Storage
+{
+    public class JsonPersistentStorage : IPersistentStorage
+    {
+        private const string JsonDataDirectory = "Resources/";
+
+        private readonly ILogger _logger;
+
+        public JsonPersistentStorage(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        private static string ToStoragePath(MemberInfo objType, string collection)
+            => Path.Combine(JsonDataDirectory, objType.Name, collection);
+
+        public void Store<T>(T obj, string collection, string key)
+        {
+            var path = ToStoragePath(typeof(T), collection);
+            var filePath = Path.Combine(path, $"{key}.json");
+            var json = JsonConvert.SerializeObject(obj);

```

Comment:
Can we maybe add the pre-proccessor again for `IF DEBUG` we use `Formatting.Indentened`

Reply:
I'm on it, chief! :+1: